nncf/experimental/quantization/quantizers/torch_ao_adapter.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from collections import defaultdict
+from typing import Dict, List, Tuple, Union
+
+import torch
+import torch.fx
+from torch.ao.quantization.quantizer import Quantizer as TorchAOQuantizer
+from torch.ao.quantization.quantizer.quantizer import QuantizationSpec
+from torch.ao.quantization.quantizer.quantizer import QuantizationSpecBase
+from torch.ao.quantization.quantizer.quantizer import SharedQuantizationSpec
+
+import nncf
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.logging import nncf_logger
+from nncf.common.quantization.quantizer_setup import ActivationQuantizationInsertionPoint
+from nncf.common.quantization.quantizer_setup import QuantizationPointBase
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizationPoint
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizerSetup
+from nncf.common.quantization.quantizer_setup import WeightQuantizationInsertionPoint
+from nncf.common.quantization.structs import QuantizationScheme as QuantizationMode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.experimental.quantization.quantizers.quantizer import Quantizer
+from nncf.experimental.torch.fx.nncf_graph_builder import GraphConverter
+
+EdgeOrNode = Union[Tuple[torch.fx.Node, torch.fx.Node]]
+
+
+class TorchAOQuantizerAdapter(Quantizer):
+    """
+    Implementation of the NNCF Quantizer interface for any given torch.ao quantizer.
+    """
+
+    def __init__(self, quantizer: TorchAOQuantizer):
+        self._quantizer = quantizer
+
+    def transform_prior_quantization(self, model: torch.fx.GraphModule):
+        self._quantizer.transform_for_annotation(model)
+
+    def get_quantization_setup(self, model: torch.fx.GraphModule, nncf_graph: NNCFGraph) -> SingleConfigQuantizerSetup:
+        original_meta = model.meta
+
+        self._quantizer.annotate(model)
```

Comment:
```suggestion
        annotated_model = self._quantizer.annotate(model)
```
https://github.com/pytorch/pytorch/blob/fd8e0e3e10785ad28398dae44368ee1d510f6ef7/torch/ao/quantization/quantizer/quantizer.py#L155

Reply:
FIY https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/quantize_pt2e.py#L97

Reply:
Done